packages/yarnpkg-pnp/sources/loader/makeApi.ts

```diff
@@ -851,11 +853,12 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
 
   function resolveRequest(request: PortablePath, issuer: PortablePath | null, {considerBuiltins, extensions, conditions, require}: ResolveRequestOptions = {}): PortablePath | null {
     const unqualifiedPath = resolveToUnqualified(request, issuer, {considerBuiltins});
-
     if (unqualifiedPath === null)
       return null;
 
-    const remappedPath = resolveUnqualifiedExport(request, unqualifiedPath, {conditions, require});
+    const remappedPath = !considerBuiltins || !builtinModules.has(request)
```

Comment:
Shouldn't the same logic also be applied when `request` is `pnpapi`?

Reply:
It can't be `pnpapi`, it has already been resolved to a file path by this point (inside `resolveToUnqualified`).

Reply:
Oh, indeed. 🤦‍♂️